app/assets/javascripts/discourse/app/routes/build-private-messages-route.js

```diff
@@ -33,18 +33,30 @@ export default (inboxType, path, filter) => {
         topicListFilter
       );
 
-      return lastTopicList
-        ? lastTopicList
-        : this.store
-            .findFiltered("topicList", { filter: topicListFilter })
-            .then((model) => {
-              // andrei: we agreed that this is an anti pattern,
-              // it's better to avoid mutating a rest model like this
-              // this place we'll be refactored later
-              // see https://github.com/discourse/discourse/pull/14313#discussion_r708784704
-              model.set("emptyState", this.emptyState());
-              return model;
-            });
+      if (lastTopicList) {
+        return lastTopicList;
+      }
+
+      params ||= {};
+      for (const [key, val] of Object.entries(params)) {
```

Comment:
I see this is twice in this PR and also repeated here:

https://github.com/discourse/discourse/blob/fb0832be3e84a12fefb4005e8ac28b94b650e255/app/assets/javascripts/discourse/app/routes/build-topic-route.js#L63-L67

I think it would be nice to make a function in `lib/utilities.js` something like this that can be continually reused on these routes:

```javascript
export function cleanNullQueryParams(params) {
  for (const [key, val] of Object.entries(params)) {
    if (val === "undefined" || val === "null") {
      params[key] = null;
    }
  }
}
```